api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -100,7 +102,11 @@ private void validateHelper(Obs obs, Errors errors, List<Obs> ancestors, boolean
 		if (obs.getObsDatetime() == null) {
 			errors.rejectValue("obsDatetime", "error.null");
 		}
-		
+		if (obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
+			if (!Objects.equals(obs.getPerson(),obs.getEncounter().getPatient().getPerson())) {
```

Comment:
```suggestion
			if (!obs.getPerson().equals(obs.getEncounter().getPatient().getPerson())) {
```

Reply:
At this point, `Objects.equals()` doesn't do anything for us.

Reply:
@ibacher true let make a couple of fixes on this, thanks